### ISSUE TITLE
Parsing $UsnJrnl files and $J data streams stored in arbitrary locations

### ISSUE THREAD

**Issue by john-corcoran** [log2timeline/plaso#1514]
**Plaso version:**

plaso-20171118

**Operating system Plaso is running on:**

Ubuntu 16.04 and Windows 7

**Installation method:**

Ubuntu: GIFT PPA (stable)
Windows: packaged release

**Description of problem:**

I'm trying to use Plaso as a parser for $UsnJrnl data stored in an arbitrary location (i.e. copied out of '/$Extend') by specifying "--parsers usnjrnl", but am having difficulty with plaso-20171118 on Ubuntu and Windows. 

I thought the problem might be that I'm copying the $J data stream to a distinct file for parsing, rather than having $J as a stream of $UsnJrnl. (I'm doing it this way due to file exporting restrictions with other tools - they can't export a 'true' $UsnJrnl file with the $J data stream intact). However I have also tried recreating a new $UsnJrnl file with a re-attached $J data stream (using https://github.com/RobinDavid/pyADS), which also isn't recognised by Plaso.

On my Ubuntu instance of Plaso, I have tried (without success) expanding the scope of line 333 in plaso/engine/extractors.py to allow for $J files:
`((filename_upper == '$USNJRNL' and data_stream_name == '$J') or (filename_upper == '$J'))):`

I also wondered if the IsMetadata checks (that use _METADATA_FILE_LOCATIONS_TSK) in plaso/engine/worker.py might be mandating that $UsnJrnl files can only be parsed if located in a $Extend folder, and therefore cannot be parsed in an arbitrary location - but I'm not sure that's the case, as from testing I see that a $MFT file can be parsed in any location despite it being listed in _METADATA_FILE_LOCATIONS_TSK as only metadata (and therefore parse-able) if located in the file system root.

So in summary, would it be possible to expand Plaso capability to include identification of $J data streams (even if they sit in a distinct file) and parse them in an arbitrary location?

**Comment by joachimmetz** [comment]
This is working as currently implemented. Plaso only parses the `$UsnJrnl:$J` from a storage media image or device. The information in this data stream does not has a clear signature so plaso (at the moment) cannot determine similar content in arbitrary files. One idea is to support [force parser mode](https://github.com/log2timeline/plaso/issues/904)

**Comment by joachimmetz** [comment]
Closing issue in favor of https://github.com/log2timeline/plaso/issues/3604. No response from original reporter. Similar request to have this functionality has been asked on the Slack channel once or twice since this issue.
